From df54ff9ded6704b7c8a1769214563ceb22359c3f Mon Sep 17 00:00:00 2001
From: Clark Wang <xiaoning.wang@nxp.com>
Date: Mon, 19 Nov 2018 10:24:11 +0800
Subject: [PATCH 5193/5242] MLK-20351-1 dtsi: imx8mm: correct the DMA channel
 for ECSPI

commit  fd2b3f61943d865e791ec284ecdd6fcd91d05051 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fix the wrong configuration of "dmas" in dts. This leads the spi
transfer error in dma mode and cause this issue.

Signed-off-by: Clark Wang <xiaoning.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
index d3669ab..4ca9cf0 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
@@ -915,7 +915,7 @@
 		clocks = <&clk IMX8MM_CLK_ECSPI1_ROOT>,
 			<&clk IMX8MM_CLK_ECSPI1_ROOT>;
 		clock-names = "ipg", "per";
-		dmas = <&sdma1 0 4 0>, <&sdma1 1 4 0>;
+		dmas = <&sdma1 0 7 1>, <&sdma1 1 7 2>;
 		dma-names = "rx", "tx";
 		status = "disabled";
 	};
@@ -929,7 +929,7 @@
 		clocks = <&clk IMX8MM_CLK_ECSPI2_ROOT>,
 			<&clk IMX8MM_CLK_ECSPI2_ROOT>;
 		clock-names = "ipg", "per";
-		dmas = <&sdma1 2 4 0>, <&sdma1 3 4 0>;
+		dmas = <&sdma1 2 7 1>, <&sdma1 3 7 2>;
 		dma-names = "rx", "tx";
 		status = "disabled";
 	};
@@ -943,7 +943,7 @@
 		clocks = <&clk IMX8MM_CLK_ECSPI3_ROOT>,
 			<&clk IMX8MM_CLK_ECSPI3_ROOT>;
 		clock-names = "ipg", "per";
-		dmas = <&sdma1 4 4 0>, <&sdma1 5 4 0>;
+		dmas = <&sdma1 4 7 1>, <&sdma1 5 7 2>;
 		dma-names = "rx", "tx";
 		status = "disabled";
 	};
-- 
1.7.9.5

